lib/improver/wind_gust_diagnostic.py

```diff
@@ -0,0 +1,232 @@
+    For example a typical wind-gust could be MAX(gust(50%),windspeed(95%))
+    an extreme wind-gust forecast could be MAX(gust(95%), windspeed(100%))
+
+    Scientific Reference: *Roberts N., Mylne K.*
+    Poster - European Meteorological Society Conference 2017.
+
+    See
+    https://github.com/metoppv/improver/files/1244828/WindGustChallenge_v2.pdf
+    for a discussion of the problem and proposed solutions.
+
+    """
+
+    def __init__(self, percentile_gust, percentile_windspeed):
+        """
+        Create a WindGustDiagnostic plugin for a given set of percentiles.
+
+        Args:
+            percentile_gust (float):
+                Percentile value required from wind-gust cube.
+            percentile_windspeed (float):
+                Percentile value required from wind-speed cube.
+
+        """
+        self.percentile_gust = percentile_gust
+        self.percentile_windspeed = percentile_windspeed
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        desc = ('<WindGustDiagnostic: wind-gust perc='
+                '{0:3.1f}, wind-speed perc={1:3.1f}>'
+                .format(self.percentile_gust, self.percentile_windspeed))
+        return desc
+
+    def add_metadata(self, cube):
+        """Add metadata to cube for windgust diagnostic.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing the wind-gust diagnostic data.
+        Returns:
+            result (iris.cube.Cube):
+                Cube containing the wind-gust diagnostic data with
+                corrected Metadata.
+
+        """
+        result = cube
+        result.standard_name = "wind_speed_of_gust"
+        result.long_name = "wind_gust_diagnostic"
+        if self.percentile_gust == 50.0 and self.percentile_windspeed == 95.0:
+            diagnostic_txt = 'Typical gusts'
+        elif (self.percentile_gust == 95.0 and
+              self.percentile_windspeed == 100.0):
+            diagnostic_txt = 'Extreme gusts'
+        else:
+            diagnostic_txt = str(self)
+        result.attributes.update({'wind_gust_diagnostic': diagnostic_txt})
+
+        return result
+
+    @staticmethod
+    def update_metadata_after_max(cube, perc_coord):
+        """Update metadata after MAX found through merged_cube.collapsed
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing the wind-gust diagnostic data.
+        Returns:
+            result (iris.cube.Cube):
+                Cube containing the wind-gust diagnostic data with
+                corrected Metadata.
+
+        """
+        result = cube
+        result.remove_coord(perc_coord)
+        return result
+
+    @staticmethod
+    def extract_percentile_data(cube, req_percentile, standard_name):
+        """Extract percentile data from cube.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube contain one or more percentiles of wind_gust data.
```

Comment:
This has been written as quite a standalone piece of code, I imagine with the idea that it could go into utilities? If this is the case please remove the reference to `wind_gust data` here, that will keep it entirely generic.